LiskHQ/lisk-nano
# Update to Lisk Core 1.0.0 API - Closes #341

### What was the problem?

Lisk Nano used Lisk Core 0.9.X API

### How did I fix it?

Updated to Lisk Core 1.0.0 API

Changes include the following:
- some endpoint names changed (e.g. `/api/accounts/delegates` to `/api/votes`)
- each response now contains the main content in `data` object, before the name of the object depended on the endpoint (e.g. `transactions`, `delegates`)
- `account.secondSignature` property was removed. `account.secondPublicKey` can be used instead
- `delegate. rate` property was removed. `delegate.rank` can be used instead
- `delegate.address` and `delegate.publicKey` was moved to `delegate.account.address` and `delegate.account.publicKey`

There were some issues in Lisk Core:

Every thing works completely with latest changes  of `lisk core` witch contains in `development` branch, but there is no release for some of these features. So I had to suspend some of e2e tests.

--  infinite scroll in all lists are broken. there is a separated task to fix them #1065 
### How to test it?

1. Checkout `development` branch of lisk core
2. Then run lisk nano and check all features

### Review checklist

- The PR solves #341
- All new code is covered with unit tests
- All new features are covered with e2e tests
- All new code follows best practices
